### PR TITLE
systemd-wait-online: improve -i handling

### PR DESCRIPTION
**Pull request by teg**
This fixes a race and changes the behavior of the -i switch slightly. It used to be that -i would mean that we wait for at least those interfaces, but now we also ignore all others.

This is to address comments first made in PR #728.

### CODE REVIEW

File: src/network/networkd-wait-online-manager.c [hunk 1]
```diff
@@ -38,9 +38,15 @@ bool manager_ignore_link(Manager *m, Link *link) {
         assert(m);
         assert(link);
 
+        /* always ignore the loopback interface */
         if (link->flags & IFF_LOOPBACK)
                 return true;
 
+        /* if interfaces are given on the commandlin, ignore all others */
```
Comment (martinpitt) [inline src/network/networkd-wait-online-manager.c:45]: "command line"

**Comment by martinpitt** [comment]
This makes sense and works fine now, thank you! Except for the typo in the comment this is ready to land from my POV.

**Comment by zonque** [comment]
Fixed the typo and pushed manually.

**Comment by poettering** [comment]
Daniel, to easy review would be nice to mention the hash id for this here when you push manually!

**Comment by zonque** [comment]
be0c1370

**Comment by poettering** [comment]
Thanks!

**Pull request closed without merging**